subsys/bluetooth/audio/ascs.c

```diff
@@ -1425,22 +1424,31 @@ static int ase_stream_qos(struct bt_audio_stream *stream,
 		}
 	}
 
-	iso = audio_iso_get_or_new(ascs, cig_id, cis_id);
-	if (iso == NULL) {
-		BT_ERR("Could not allocate audio_iso");
-		return -ENOMEM;
+	/* QoS->QoS transition. Unbind ISO if CIG/CIS changed. */
+	if (ep->iso != NULL && ep->cig_id != cig_id && ep->cis_id != cis_id) {
+		bt_audio_iso_unbind_ep(ep->iso, ep);
```

Comment:
Shouldn't this be
```suggestion
	if (ep->iso != NULL && (ep->cig_id != cig_id || ep->cis_id != cis_id)) {
		bt_audio_iso_unbind_ep(ep->iso, ep);
```

If either the CIG or CIS ID gets modified when the audio_iso is bound, then we need to rebind. 

With the current check, the audio_iso would not be unbound if
Current CIG_ID=2, CIS_ID=5
New CIG_ID=2, CIS_ID7 
As the CIG_ID didn't change.

Reply:
I should be more careful pushing patches on Friday's evening :D